File: backend/app/v1/schemas/interview.py
"""
Pydantic schemas for interview management.
"""

from __future__ import annotations

import uuid
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict


class InterviewCreate(BaseModel):
    """Schema for creating a new interview session."""

    candidate_id: uuid.UUID
    job_id: uuid.UUID


class InterviewRead(BaseModel):
    """Schema for reading interview details."""

    id: uuid.UUID
    candidate_id: uuid.UUID
    job_id: uuid.UUID
    interviewer_id: uuid.UUID
    status: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class InterviewDecision(BaseModel):
    """Schema for HR decision after reviewing interview results."""

    decision: Literal["proceed", "reject"]
    notes: str | None = None


class InterviewListResponse(BaseModel):
    """Schema for listing interviews."""

    interviews: list[InterviewRead]
    total: int

File: frontend/src/apis/types/interview.ts
export type InterviewStatus = "pending" | "completed" | "rejected" | "cancelled";

export interface InterviewRead {
  id: string;
  candidate_id: string;
  job_id: string;
  interviewer_id: string;
  status: InterviewStatus;
  created_at: string;
}

export interface InterviewListResponse {
  interviews: InterviewRead[];
  total: number;
}

export interface InterviewCreate {
  candidate_id: string;
  job_id: string;
}

export interface InterviewDecision {
  decision: "proceed" | "reject";
  notes?: string;
}
